apps/src/code-studio/assets/SpriteUpload.js

```diff
@@ -4,15 +4,51 @@ export default class SpriteUpload extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      file: null
+      fileData: null,
+      filePreviewURL: '',
+      filename: '',
+      category: ''
     };
   }
 
-  handleSubmit(event) {}
+  handleSubmit = event => {
+    event.preventDefault();
+    let xhr = new XMLHttpRequest();
 
-  handleChange = event => {
+    const onError = function() {
+      console.log(new Error(`${xhr.status} ${xhr.statusText}`));
+    };
+
+    const onSuccess = function() {
+      if (xhr.status >= 400) {
+        onError();
+        return;
+      }
+    };
+
+    xhr.addEventListener('load', onSuccess);
+    xhr.addEventListener('error', onError);
+    xhr.open(
+      'POST',
+      `/api/v1/animation-library/level_animations/${this.state.filename}`,
+      true
+    );
+    xhr.setRequestHeader('Content-type', 'image/png');
+    xhr.send(this.state.fileData);
+  };
+
+  handleImageChange = event => {
+    let file = event.target.files[0];
+    this.setState({
+      fileData: file,
+      filename: file.name,
+      filePreviewURL: URL.createObjectURL(file)
+    });
+  };
+
+  handleCategoryChange = event => {
     this.setState({
-      file: URL.createObjectURL(event.target.files[0])
+      category: event.target.value
```

Comment:
Thanks for pointing this out! I have this change in my next PR where we use the category to given to make the URL. This also points out a limitation in my plan if a curriculum writer wants to create a new category.

Reply:
yeah, i don't think it would be possible for them to create new categories right now (without at least moving the animation previews to S3). we could either:

1. plan that work if it's a requirement that curriculum can create new categories, or
2. pull the list of categories (i'm not sure where the best place would be -- maybe just S3?) and display it as a dropdown instead of a text input